src/QMCWaveFunctions/Fermion/DiracDeterminantBatched.cpp

```diff
@@ -664,8 +664,9 @@ typename DiracDeterminantBatched<DET_ENGINE>::GradType DiracDeterminantBatched<D
   {
     resizeScratchObjectsForIonDerivs();
     Phi->evaluateGradSource(P, FirstIndex, LastIndex, source, iat, grad_source_psiM);
-    // FIXME due padding of psiMinv
-    g = simd::dot(psiMinv.data(), grad_source_psiM.data(), psiMinv.size());
+    // psiMinv.cols() can be different from NumOrbitals
+    for (int i = 0; i < psiMinv.rows(); i++)
+      g += simd::dot(psiMinv[i], grad_source_psiM[i], NumOrbitals);
```

Comment:
The reason this is necessary is that grad_source_psiM does not have explicit padding and so doesn't have aligned rows, which seems like it could be a performance regression. SIMD::dot does not actually do the transpose of A but actually assumes A has been transposed.  If both A and B have the same row alignment then simple element wise product and accumulation is fine and likely optimal for vectorization. 

But if grad_source_psiM has unaligned rows by design and not omission then a unit test should capture this special behavior.  Because at face value taking an inner produce of a {P,N} matrix and a {N,N} matrix with P>N is nonsense.

Reply:
Will add more verbose comment.

Reply:
psiMinv documentation added.